include/soci/rowset.h

```diff
@@ -41,10 +41,6 @@ class rowset_iterator
     rowset_iterator(statement & st, T & define)
         : st_(&st), define_(&define)
     {
-        assert(0 != st_);
```

Comment:
While I'm convinced by the reasoning in #300:
> application shouldn't necessarily crash just because some database operation failed
I'm not sure removal of assertions checking the class invariants like this and the following one fall into this category. The `rowset_iterator` must not contain nullptr reference to statement or define and that assertion documents this fact.